client/src/components/ui/title.tsx

```diff
@@ -0,0 +1,27 @@
+import { FC, PropsWithChildren } from "react";
+
+import { cn } from "@/lib/utils";
+
+const styles = {
+  lg: "text-lg leading-10",
+  xl: "text-xl leading-12",
+  "2xl": "text-2xl leading-16",
+};
+
+const Title: FC<
+  PropsWithChildren<{
+    as: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+    size?: keyof typeof styles;
+    className?: HTMLHeadingElement["className"];
```

Comment:
Just curious, what is the benefit of using this notation instead of saying type string?

Reply:
Not a real benefit but more like pointing to source of truth: while it's obvious className is a `string`, and you could type as if, I prefer making references to the "source" component they come from. In the end all HTML elements share common types so I prefer point to them and, in a hypothetic case in the future `className` is not a `string` anymore, I don't need to touch my code, only the typescript will need to be updated (specifically the DOM types).

if my code relies on other code, I try to reference to it, so if that code changes, I will notice the types failing on my code. 

For example here: https://github.com/Vizzuality/4-growth/blob/902619fa576a4da188c5a693366538644bbca0db/client/src/containers/profile/saved-visualizations/table/columns/cell-name/index.tsx#L41-L43 

The parameter of that function is an event triggered when a user presses a key, while I could come up with my own typing for this, I know that event comes from an Input, and that Input is triggering a `onKeyDown` so I preferrer referring to their types instead of coming up with my own.

Not sure if the explanation is clear, otherwise, let me know and we can talk about it.